esmvalcore/preprocessor/_time.py

```diff
@@ -261,18 +282,24 @@ def seasonal_statistics(cube, operator='mean'):
         Seasonal statistic cube
     """
     if not cube.coords('clim_season'):
-        iris.coord_categorisation.add_season(cube, 'time', name='clim_season')
+        iris.coord_categorisation.add_season(
+            cube, 'time', name='clim_season', seasons=seasons
+        )
     if not cube.coords('season_year'):
         iris.coord_categorisation.add_season_year(
-            cube, 'time', name='season_year')
+            cube, 'time', name='season_year', seasons=seasons
+        )
 
     operator = get_iris_analysis_operation(operator)
 
     cube = cube.aggregated_by(['clim_season', 'season_year'],
                               operator)
 
+    if any(len(season) != 3 for season in seasons):
```

Comment:
This seems to contradict the docstring above:

> Chunks time in 3-month periods and computes statistics over them;

Also, is the `cube.extract` function below not needed for seasons different from 3 months? I think it's there mainly to prevent 'incomplete' seasons..